azure-pipelines.yml

```diff
@@ -49,3 +49,71 @@ jobs:
       testRunTitle: 'Publish test results for $(Agent.JobName) $(TEST_MODE) $(PYTHON_VERSION)'
       failTaskOnFailedTests: true
     condition: succeededOrFailed()
+
+- job:
+  displayName: Linux
+  pool:
+    vmImage: 'ubuntu-18.04'
+  strategy:
+    matrix:
+      ubuntu_python36:
+        DISTRIB: 'ubuntu'
+        PYTHON_VERSION: '3.6'
+        SPHINXOPTS: ''
+      conda_python35:
+        DISTRIB: 'conda'
+        PYTHON_VERSION: '3.5'
+        SPHINX_VERSION: '1.8.3'
+      conda_python36:
+        DISTRIB: 'conda'
+        PYTHON_VERSION: '3.6'
+      conda_python36_localeC:
+        DISTRIB: 'conda'
+        PYTHON_VERSION: '3.6'
+        LOCALE: 'C'
+      conda_python37_localeC:
+        DISTRIB: 'conda'
+        PYTHON_VERSION: '3.7'
+        LOCALE: 'C'
+        SPHINXOPTS: '-nWT --keep-going'
+      conda_python38_sphinxDev:
+        DISTRIB: 'conda'
+        PYTHON_VERSION: '3.8'
+        SPHINX_VERSION: 'dev'
+      python37_minimal:
+        DISTRIB: 'minimal'
+        PYTHON_VERSION: '3.7'
+      # pythonNightly:
+      #   DISTRIB: 'Nightly'
+  condition: not(contains(variables['Build.SourceVersionMessage'], '[doc only]'))
```

Comment:
This also doesn't work

Reply:
I wouldn't worry about `[doc only]` but instead use standard `[ci skip]` and what should someday be supported `[skip azp]` and even then it probably won't be used too often since our CIs are fast. You can use this code:

https://github.com/mne-tools/mne-python/blob/master/azure-pipelines.yml#L16-L34

Reply:
If I got the right page, supposedly [ci skip] is meant to work already: https://github.com/Microsoft/azure-pipelines-agent/issues/858#issuecomment-475768046
I'll test it out.

Reply:
Okay I see Guillaume made it work in mne: https://github.com/mne-tools/mne-python/pull/8514/

but what does $RET mean and how is it used downstream?
```
- stage: Check
  jobs:
    - job: Skip
      pool:
        vmImage: 'ubuntu-18.04'
      variables:
        RET: 'true'
      steps:
      - bash: |
          git_log=`git log --max-count=1 --skip=1 --pretty=format:"%s"`
          echo "##vso[task.setvariable variable=log]$git_log"
      - bash: echo "##vso[task.setvariable variable=RET]false"
        condition: or(contains(variables.log, '[skip azp]'), contains(variables.log, '[azp skip]'), contains(variables.log, '[skip ci]'), contains(variables.log, '[ci skip]'))
      - bash: echo "##vso[task.setvariable variable=start_main;isOutput=true]$RET"
        name: result
```

Reply:
RET defaults to `true`, but on the second `bash` line gets turned to `false` under some conditions. Then this `RET` env var gets set to an Azure runtime variable `result` in the last step via a magic `echo` command, and this `result` Azure env var is available to subsequent steps.

Reply:
Sorry I was more asking what does RET stand for? Does this variable name have a special meaning?

Reply:
It's just an all-upper (BASH typical naming) shorthand for "return" I think, could be some other name (but I'd keep it this one so that copy-paste comparisons to MNE make sense)